### PR TITLE
Try support FSharp 7

### PR DESCRIPTION
**Pull request by rstm-sf**

### CODE REVIEW

**Comment by ashmind** [comment]
@rstm-sf This is in draft, but since the build actually passed, is F# part ready to merge?
Seems like conflicts are related to ref assembly fixes, but if it's slimmed down to F# should avoid those.

**Comment by alrz** [comment]
Was just looking for this. Would love to see it get merged soon :)

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -23,6 +23,9 @@ jobs:
     - uses: actions/setup-dotnet@v1
       with:
           dotnet-version: 6.0.x
+    - uses: actions/setup-dotnet@v1
+      with:
+          dotnet-version: 7.0.x
```
Comment (ForNeVeR) [inline .github/workflows/dotnet.yml:28]: FYI, it supports multiple versions (at least in `v3`, I'm not that sure about `v1`).
```suggestion
          dotnet-version: |
            6.0.x
            7.0.x
```

**Comment by rstm-sf** [comment]
Sorry for delay.

FCS used to depend on msbuild (https://github.com/dotnet/msbuild/tree/main/src/Utilities#netstandard20-target), but the dependency has been removed from the 43.7.200 version https://www.nuget.org/packages/FSharp.Compiler.Service/43.7.200#dependencies-body-tab

**Comment by baronfel** [comment]
Having done these in the past, this one LGTM 👍

**Comment by Tangent-90** [comment]
when will this be merged?

**Comment by ashmind** [comment]
Thanks for your work! Unfortunately I wasn't able to support it promptly enough due to other priorities.
However now that https://github.com/ashmind/mirrorsharp/pull/223 is merged, it should cover the newest F#.

Therefore I'll close this one.

**Pull request closed without merging**